packages/application-shell-connectors/src/components/application-context/application-context.tsx

```diff
@@ -153,7 +153,16 @@ const ApplicationContext = <AdditionalEnvironmentProperties extends {}>(
   props: ConsumerProps<AdditionalEnvironmentProperties>
 ) => (
   <Context.Consumer>
-    {applicationContext => props.render(applicationContext)}
+    {context => {
+      // Because of the way the ApplicationShell configures the Context.Provider,
+      // we ensure that, when we read from the context, we always get actual
+      // context object and not the initial value.
+      // Therefore, we can safely cast the value to be out `TApplicationContext` type.
+      const applicationContext = context as TApplicationContext<
```

Comment:
So the alternative (not really) would be consumers using type guards to ensure it is not an empty object?

Reply:
Probably yes, because you then always get `TApplicationContext | {}` which it would never be the case given our setup.